### PR TITLE
Add parental rating fields to EPG, Timers and Recordings.

### PR DESCRIPTION
**Pull request by DeltaMikeCharlie**
v22.0.0

Add parental rating fields in support of [Kodi PR #24096](https://github.com/xbmc/xbmc/pull/24096)

Access via $INFO[ListItem.ParentalRating], $INFO[ListItem.ParentalRatingCode], $INFO[ListItem.ParentalRatingIcon] and $INFO[ListItem.ParentalRatingSource] respectively.

Add ability to populate existing cast, director and writer fields.  Access via existing info labels.

Note: 'ListItem.ParentalRatingCode' will be available once [Kodi PR 23586](https://github.com/xbmc/xbmc/pull/23586) has been merged.  However, the columns are already in the epgtags table of the PVR/EPG database.

### CODE REVIEW

**Comment by DeltaMikeCharlie** [comment]
@phunkyfish - I received an email notification that this build had failed after you appear to have manually re-triggered it:  "Build and run tests, Attempt #2: All jobs have failed"

I've also seen this message on GitHub: "This request was automatically failed because there were no enabled runners online to process the request for more than 1 days."

This seems to me to be an issue with the build system.  However, can you please let me know if this was caused by something that I missed when submitting my PR?

**Comment by ksooo** [comment]
Two comments:
* as you are adding a new feature with this release, please increase addon minor version number, not micro 
* I would use different combinations for the entries, some with both new values, some with only one, some without. Pvr.demo is more for demonstration of features and testing. That way for example we can easily see how different combinations are presented in Estuary.

File: src/PVRDemo.h [hunk 1]
```diff
@@ -24,13 +24,14 @@ struct PVRDemoEpgEntry
   int iGenreType;
   int iGenreSubType;
   //  time_t firstAired;
-  //  int iParentalRating;
+  int iParentalRating;
```
Comment (ksooo) [inline src/PVRDemo.h]: Please default init the variable here.

Comment (DeltaMikeCharlie) [reply src/PVRDemo.h]: Complete.

**Comment by DeltaMikeCharlie** [comment]
> Two comments:
> 
>     * as you are adding a new feature with this release, please increase addon minor version number, not micro
> 
>     * I would use different combinations for the entries, some with both new values, some with only one, some without. Pvr.demo is more for demonstration of features and testing. That way for example we can easily see how different combinations are presented in Estuary.

Updated as requested.

**Comment by DeltaMikeCharlie** [comment]
The previous force-push was an update to accommodate https://github.com/xbmc/xbmc/pull/24096/ (adding parental rating icons).

**Comment by ksooo** [comment]
As I said in the pvr.hts PR, merge of this must wait until P* dev cycle has started, because of the Addon-API change which will not be part of Omega.

File: pvr.demo/addon.xml.in [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <addon
   id="pvr.demo"
-  version="21.0.0"
+  version="21.2.0"
```
Comment (ksooo) [inline pvr.demo/addon.xml.in]: That will definitely end up with some 22.x.y number

**Comment by ksooo** [comment]
We cannot merge this for Omega, due to API incompatibility.

**Pull request closed without merging**